src/node_buffer.cc

```diff
@@ -602,6 +602,83 @@ void Compare(const FunctionCallbackInfo<Value> &args) {
 }
 
 
+void IndexOfString(const FunctionCallbackInfo<Value>& args) {
+  ASSERT(args[0]->IsObject());
+  ASSERT(args[1]->IsString());
+  ASSERT(args[2]->IsNumber());
+
+  ARGS_THIS(args[0].As<Object>());
+  node::Utf8Value str(args.GetIsolate(), args[1]);
+  uint32_t offset = args[2]->Uint32Value();
+
+  if (str.length() == 0) {
+    return args.GetReturnValue().Set(
+        MIN(static_cast<uint32_t>(obj_length), offset));
+  }
+  if (str.length() + offset > obj_length || obj_length == 0) {
+    return args.GetReturnValue().Set(-1);
+  }
+
+  char* result = strstr(obj_data + offset, *str);
+  args.GetReturnValue().Set(
+      result ? static_cast<int32_t>(result - obj_data) : -1);
+}
+
+
+void IndexOfBuffer(const FunctionCallbackInfo<Value>& args) {
+  ASSERT(args[0]->IsObject());
+  ASSERT(args[1]->IsObject());
+  ASSERT(args[2]->IsNumber());
+
+  ARGS_THIS(args[0].As<Object>());
+  Local<Object> buf = args[1].As<Object>();
+  uint32_t offset = args[2]->Uint32Value();
+  size_t buf_length = buf->GetIndexedPropertiesExternalArrayDataLength();
+  char* buf_data =
+      static_cast<char*>(buf->GetIndexedPropertiesExternalArrayData());
+  if (buf_length > 0)
+    CHECK_NE(buf_data, nullptr);
+
+  if (buf_length + offset > obj_length || obj_length == 0)
```

Comment:
Guess I hadn't considered that. So will the result of `buf_length + offset` be cast to a `uint32_t`?